device/ethernet.yaml

```diff
@@ -1,58 +1,85 @@
-            The Link Local IPv6 address will be automatically generated.
-          type: array
-          items:
-            $ref: './ipv6.yaml#/components/schemas/Device.Ipv6'
-      
\ No newline at end of file
+components:
+  schemas:
+    Device.EthernetBase:
+      x-include: 
+      - '../common/common.yaml#/components/schemas/Named.Object'
+      description: >-
+        Base Ethernet interface.
+      type: object
+      required: [mac]
+      properties:
+        mac:
+          description: >-
+            Media Access Control address.
+          type: string
+          format: mac
+        mtu:
+          description: >-
+            Maximum Transmission Unit.
+          type: integer
+          minimum: 0
+          maximum: 65535
+          default: 1500
+        vlans:
+          description: >-
+            List of VLANs
+          type: array
+          items:
+            $ref: './vlan.yaml#/components/schemas/Device.Vlan'
+
+    Device.Ethernet:
+      x-include: 
+      - '#/components/schemas/Device.EthernetBase'
+      description: >-
+        An Ethernet interface with IPv4 and IPv6 addresses.
+      type: object
+      properties:
+        port_name: 
+          description: |-
+            The unique name of a Port or a LAG that will emulate this interface.
+          x-status: deprecated
+          type: string
+          x-constraint:
+          - '/components/schemas/Port/properties/name'
+          - '/components/schemas/Lag/properties/name'
+        connection: 
+          description: |-
+            Device connection to physical, LAG or another device.
+          $ref : '#/components/schemas/Device.Connection'
+        ipv4_addresses:
+          description: |-
+            List of IPv4 addresses and their gateways. 
+          type: array
+          items:
+            $ref: './ipv4.yaml#/components/schemas/Device.Ipv4' 
+        ipv6_addresses:
+          description: |-
+            List of global IPv6 addresses and their gateways.
+            The Link Local IPv6 address will be automatically generated.
+          type: array
+          items:
+            $ref: './ipv6.yaml#/components/schemas/Device.Ipv6'
+      
+    Device.Connection:
+      description: Device connection to physical port, LAG port or another device.
+      type: object
+      properties:
+        choice:
+          description: port_name, lag_name or device_name
+          type: string
+          enum: [port_name, lag_name, vxlan_name]
+        port_name:
+          description: Name of the port that the device is connected to.
+          type: string
+          x-constraint:
+          - '/components/schemas/Port/properties/name'
+        lag_name:
+          description: Name of the Lag the device is connected to.
+          type: string
+          x-constraint:
+          - '/components/schemas/Lag/properties/name'
+        vxlan_name:
+          description: Name of the vxlan that acts as overlay for the device.

```

Comment:
Name of the VxLAN instance ( or VxLAN tunnel )  that this Ethernet interface is connected to.

Reply:
updated